react/features/shared-video/functions.ts

```diff
@@ -70,20 +76,71 @@ export function extractYoutubeIdOrURL(input: string) {
         return;
     }
 
-    const youtubeId = getYoutubeId(trimmedLink);
+    if (areYoutubeURLsAllowedForSharedVideo()) {
+        const youtubeId = getYoutubeId(trimmedLink);
 
-    if (youtubeId) {
-        return youtubeId;
+        if (youtubeId) {
+            return youtubeId;
+        }
     }
 
     // Check if the URL is valid, native may crash otherwise.
     try {
         // eslint-disable-next-line no-new
-        new URL(trimmedLink);
+        const url = new URL(trimmedLink);
+
+        if (!url?.hostname || !URL_WHITELIST.includes(url?.hostname)) {
+            return;
+        }
     } catch (_) {
         return;
     }
 
     return trimmedLink;
 }
 
+/**
+ * Returns true if shared video functionality is enabled and false otherwise.
+ *
+ * @param {IStateful} stateful - - The redux store or {@code getState} function.
+ * @returns {boolean}
+ */
+export function isSharedVideoEnabled(stateful: IStateful) {
+    const state = toState(stateful);
+    const { disableThirdPartyRequests = false } = state['features/base/config'];
+
+    return !disableThirdPartyRequests && URL_WHITELIST.length > 0;
+}
+
+/**
+ * Checks if you youtube URLs should be allowed for shared videos.
+ *
+ * @returns {boolean}
+ */
+export function areYoutubeURLsAllowedForSharedVideo() {
+    return URL_WHITELIST.includes(YOUTUBE_URL_DOMAIN);
+}
+
+/**
+ * Returns true if the passed url is allowed to be used for shared video or not.
+ *
+ * @param {string} url - The URL.
+ * @returns {boolean}
+ */
+export function isURLAllowedForSharedVideo(url: string) {
+    if (!url) {
+        return false;
+    }
+
+    if (!url.match(/http/)) {
```

Comment:
You want to match from the start `^` but allso what about casing? `hTTpS` ? Isn't is better to just parse the URL all the time and act on the exception?